src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceIntTest.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.digital.ho.pttg.api;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlGroup;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.digital.ho.pttg.application.ServiceConfiguration;
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.AFTER_TEST_METHOD;
+import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.BEFORE_TEST_METHOD;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@SqlGroup({
+        @Sql(executionPhase = BEFORE_TEST_METHOD, scripts = "/sql/ArchiveResourceIntTest/before.sql"),
+        @Sql(executionPhase = AFTER_TEST_METHOD, scripts = "/sql/ArchiveResourceIntTest/after.sql")
+})
+public class ArchiveResourceIntTest {
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Before
+    public void setUp() {
+        ServiceConfiguration serviceConfiguration = new ServiceConfiguration(objectMapper, 0, 0);
+        objectMapper = ReflectionTestUtils.invokeMethod(serviceConfiguration, "initialiseObjectMapper", objectMapper);
+    }
+
+    @Test
+    public void shouldReturnArchivedResults() throws JsonProcessingException {
+        List<ArchivedResult> expected = asList(
+                new ArchivedResult(ImmutableMap.of("PASS", 5, "FAIL", 3)),
+                new ArchivedResult(ImmutableMap.of("ERROR", 2, "NOTFOUND", 99))
+        );
+
+        String archivedResults = restTemplate.getForObject("/archived?fromDate={fromDate}&toDate={toDate}", String.class, ImmutableMap.of(
+                "fromDate", LocalDate.of(2018, Month.SEPTEMBER, 1),
+                "toDate", LocalDate.of(2018, Month.SEPTEMBER, 30)
```

Comment:
Would there be a benefit in making at least one or both of the dates on the boundary of the test data?

Reply:
Yes - I have adjusted the data to improve the test